scene/gui/control.cpp

```diff
@@ -2017,6 +2017,33 @@ bool Control::has_focus() const {
 	return is_inside_tree() && get_viewport()->_gui_control_has_focus(this);
 }
 
+bool Control::child_has_focus(const Control *p_from) const {
```

Comment:
I'm not sure about this function, it's only used for `ScrollContainer`, so maybe you can keep it inside `ScrollContainer` instead and to wrap it inside `#ifdef TOOLS_ENABLED`.

Reply:
Also you can use [gui-get-focus-owner](https://docs.godotengine.org/en/stable/classes/class_viewport.html#class-viewport-method-gui-get-focus-owner) and [is_ancestor_of](https://docs.godotengine.org/en/stable/classes/class_node.html#class-node-method-is-ancestor-of) to check if the focus owner is a child of this node.